### PR TITLE
Transaction part - 2

### PR DESCRIPTION
**Pull request by sadekujjaman**

### CODE REVIEW

File: database/transaction.js [hunk 1]
```diff
@@ -1,30 +1,67 @@
...
+//TODO : chect integrity
+/**
+ * The function updates the balances of the sender and receiver accounts and then updates their account
+ * information.
+ */
 async function postTransaction({ sender, receiver, amount }) {
-  sender.amount -= amount;
-  receiver.amount += amount;
+  sender.balance -= amount;
+  receiver.balance += amount;
```
Comment (sadekujjaman) [inline database/transaction.js:40]: Good catch!!

File: database/transaction.js [hunk 2]
```diff
@@ -1,30 +1,67 @@
...
+/**
+ * The function checks if the provided admin ID and password match the actual admin ID and password
+ * stored in the environment variables.
+ * @returns a boolean value - `true` if the `adminId` and `adminPassword` passed as arguments match the
+ * values of `process.env.ADMIN_ID` and `process.env.ADMIN_PASSWORD` respectively, and `false`
+ * otherwise.
+ */
+function isValidAdmin({ adminId, adminPassword }) {
+  const actual_adminId = process.env.ADMIN_ID;
+  const actual_adminPassword = process.env.ADMIN_PASSWORD;
```
Comment (sadekujjaman) [inline database/transaction.js:54]: Nice!

File: index.js [hunk 3]
```diff
@@ -1,153 +1,92 @@
...
 app.post("/account", async (req, res) => {
-  const name = req.body.name;
-  const password = req.body.password;
-  let account_id = uuid.v4();
-  // TODO: use createAccount from /database/account
-  const { data, error } = await supabase.from("Account").insert({
-    account_id: account_id,
-    id: cur_id++,
-    created_at: new Date().toJSON(),
-    password: password,
```
Comment (sadekujjaman) [inline index.js:24]: Nice refactoring!

File: index.js [hunk 4]
```diff
@@ -1,153 +1,92 @@
...
-    };
-    res.json(response);
-  } else {
-    res.send("no such id exists");
+//fetch account by id:
+app.get("/account/:account_id", async (req, res) => {
+  const account_id = req.params.account_id;
+  const data = await fetchAccountById(account_id);
+  if (data) {
+    res.send(data);
```
Comment (sadekujjaman) [inline index.js]: `res.json(data)` here?

File: index.js [hunk 5]
```diff
@@ -1,153 +1,92 @@
...
-  } else {
-    res.send("No such account exist :D");
+//transaction management :
+app.post("/transaction", async (req, res) => {
+  const {
+    amount,
+    sender_id: senderId,
+    receiver_id: receiverId,
+    password,
+  } = req.body;
```
Comment (sadekujjaman) [inline index.js:63]: Very nice.

File: index.js [hunk 6]
```diff
@@ -1,153 +1,92 @@
...
+  try {
+    await makeTransaction({
+      amount,
+      receiverId,
+      senderId,
+      senderPassword: password,
+    });
```
Comment (sadekujjaman) [inline index.js:70]: How about returning a success message here?

**Review by sadekujjaman — COMMENTED** [review]
Looks very good. Nice refactoring and well-documented! Thank you. Added a couple of comments for minor changes. Feel free to adjust or merge.

File: database/transaction.js (hunk 1 above)
Comment (shimlakanu) [reply database/transaction.js:40]: thanks for your appreciation!

File: index.js (hunk 4 above)
Comment (shimlakanu) [reply index.js]: resolved

File: index.js (hunk 6 above)
Comment (shimlakanu) [reply index.js:70]: have added a message here

**Pull request merged**